Code/Source/RobotImporter/URDF/VisualsMaker.cpp

```diff
@@ -0,0 +1,174 @@
+        auto geometry = visual->geometry;
+        switch (geometry->type)
+        {
+        case urdf::Geometry::SPHERE:
+            {
+                auto sphereGeometry = std::dynamic_pointer_cast<urdf::Sphere>(geometry);
+                entity->CreateComponent(LmbrCentral::EditorSphereShapeComponentTypeId);
+                entity->Activate();
+                LmbrCentral::SphereShapeComponentRequestsBus::Event(
+                    entityId, &LmbrCentral::SphereShapeComponentRequests::SetRadius, sphereGeometry->radius);
+                entity->Deactivate();
+            }
+            break;
+        case urdf::Geometry::CYLINDER:
+            {
+                auto cylinderGeometry = std::dynamic_pointer_cast<urdf::Cylinder>(geometry);
+                entity->CreateComponent(LmbrCentral::EditorCylinderShapeComponentTypeId);
+                entity->Activate();
+                LmbrCentral::CylinderShapeComponentRequestsBus::Event(
+                    entityId, &LmbrCentral::CylinderShapeComponentRequests::SetRadius, cylinderGeometry->radius);
+                LmbrCentral::CylinderShapeComponentRequestsBus::Event(
+                    entityId, &LmbrCentral::CylinderShapeComponentRequests::SetHeight, cylinderGeometry->length);
+                entity->Deactivate();
+            }
+            break;
+        case urdf::Geometry::BOX:
+            {
+                auto boxGeometry = std::dynamic_pointer_cast<urdf::Box>(geometry);
+                entity->CreateComponent(LmbrCentral::EditorBoxShapeComponentTypeId);
+                AZ::Vector3 boxDimensions = URDF::TypeConversions::ConvertVector3(boxGeometry->dim);
+                entity->Activate();
+                LmbrCentral::BoxShapeComponentRequestsBus::Event(
+                    entityId, &LmbrCentral::BoxShapeComponentRequests::SetBoxDimensions, boxDimensions);
+                entity->Deactivate();
+            }
+            break;
+        case urdf::Geometry::MESH:
+            {
+                auto meshGeometry = std::dynamic_pointer_cast<urdf::Mesh>(geometry);
+
+                // TODO - a PoC solution, replace with something generic, robust, proper
+                std::filesystem::path modelPath(m_modelPath.c_str());
+                modelPath = modelPath.remove_filename();
+                auto relativePathToMesh = std::regex_replace(meshGeometry->filename, std::regex("package://"), "");
+                modelPath += relativePathToMesh;
+
+                // Get asset path for a given model path
+                auto assetPath = PrefabMakerUtils::GetAssetPathFromModelPath(modelPath);
+
+                entity->CreateComponent(AZ::Render::EditorMeshComponentTypeId);
+                entity->Activate();
+                AZ::Render::MeshComponentRequestBus::Event(
+                    entityId, &AZ::Render::MeshComponentRequestBus::Events::SetModelAssetPath, assetPath.c_str());
+                entity->Deactivate();
+
+                // TODO apply scale
```

Comment:
Nitpick: I believe this TODO no longer applies.

Reply:
It does - it is a different scale, specified within urdf (not the mesh). The mesh resource can be rescaled. Example:
` <mesh filename="package://turtlebot4_description/meshes/rplidar.dae" scale="1 1 1"/>`